server/src/main/java/org/elasticsearch/index/mapper/MappingParser.java

```diff
@@ -73,19 +73,36 @@ private static String getRemainingFields(Map<?, ?> map) {
         return remainingFields.toString();
     }
 
-    @SuppressWarnings("unchecked")
-    Mapping parse(@Nullable String type, CompressedXContent source) throws MapperParsingException {
+    static Map<String, Object> convertToMap(CompressedXContent source) {
         Objects.requireNonNull(source, "source cannot be null");
-        Map<String, Object> mapping = XContentHelper.convertToMap(source.compressedReference(), true, XContentType.JSON).v2();
-        if (mapping.isEmpty()) {
+        return XContentHelper.convertToMap(source.compressedReference(), true, XContentType.JSON).v2();
+    }
+
+    Mapping parse(@Nullable String type, CompressedXContent source) throws MapperParsingException {
+        Map<String, Object> mapping = convertToMap(source);
+        return parse(type, mapping);
+    }
+
+    /**
+     * A method to parse mapping from a source in a map form, that allows to specify explicit/implicit {@code subobjects} configuration.
```

Comment:
Similarly here, we can remove the doc about subobjects

Reply:
But keep the javadoc in general, thanks for adding it!

Reply:
👍